File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int p1 = 0;
        int p2 = 0;
        int p3 = 0;
        
        while (p3 < goal.length) {
            String word = goal[p3++];
            if (p1 < cards1.length && cards1[p1].equals(word)) {
                p1++;
            } else if (p2 < cards2.length && cards2[p2].equals(word)) {
                p2++;
            } else {
                return "No";
            }
        }
        return "Yes";
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let p1 = 0;
    let p2 = 0;
    let ans = 'Yes';
    
    goal.forEach(e => {
        if (p1 < cards1.length && e === cards1[p1]) p1++;
        else if (p2 < cards2.length && e === cards2[p2]) p2++;
        else ans = 'No';
    });
    
    return ans;
}